src/lib/operators/table_scan/abstract_table_scan_impl.hpp

```diff
@@ -25,17 +25,18 @@ class AbstractTableScanImpl {
    */
 
   template <bool CheckForNull, typename BinaryFunctor, typename LeftIterator>
-  static void __attribute__((noinline))
-  _scan_with_iterators(const BinaryFunctor func, LeftIterator left_it, const LeftIterator left_end,
+  static void _scan_with_iterators(const BinaryFunctor func, LeftIterator left_it, const LeftIterator left_end,
                        const ChunkID chunk_id, PosList& matches_out) {
     // Can't use a default argument for this because default arguments are non-type deduced contexts
     auto false_type = std::false_type{};
     _scan_with_iterators<CheckForNull>(func, left_it, left_end, chunk_id, matches_out, false_type);
   }
 
   template <bool CheckForNull, typename BinaryFunctor, typename LeftIterator, typename RightIterator>
-  // noinline reduces compile time drastically
-  static void __attribute__((noinline))
+  // This is a function that is critical for our performance. We want the compiler to try its best in optimizing it.
+  // Also, we want all functions called inside to be inlined (flattened) and the function itself being always aligned
+  // at a page boundary. Finally, it should not be inlined because that might break the alignment.
+  static void __attribute__((hot, flatten, aligned(256), noinline))
```

Comment:
I assume aligned(256) means we now have a 150MB release binary?

Reply:
If Hyrise only consisted of the scan, yes. ;)
39M now.